File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    # n을 이진수로 치환했을 때 1의 개수
    ocnt=str(bin(n)).count('1')
    r=n
    while True:
        r+=1
        if str(bin(r)).count('1')==ocnt:
            break
    return r

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int r=0,cnt,st=0;
    for(int i=7;i>=0;i--) {
            r=(n>>i)&1;
            if(r==1)
                st+=1;
    }
    while(true) {
        cnt=0,n+=1;
        for(int i=7;i>=0;i--) {
            r=(n>>i)&1;
            if(r==1)
                cnt+=1;
        }
        if(cnt==st) {
            r=n;
            break;
        }
    }
    return r;
}